pkg/utils/etcdutil/etcdutil.go

```diff
@@ -399,6 +404,16 @@ func (checker *healthyChecker) update(eps []string) {
 		}
 		checker.addClient(ep, time.Now())
 	}
+
+	// check if there are some stale clients, if exists, remove them.
+	checker.Range(func(key, value interface{}) bool {
+		ep := key.(string)
+		if _, ok := epMap[ep]; !ok {
+			log.Info("remove stale etcd client", zap.String("endpoint", ep))
+			checker.removeClient(ep)
+		}
+		return true
+	})
 }
```

Comment:
If we delete these nodes immediately, it will affect the speed at which servers that are unavailable for a short period of time can be rejoined.

Reply:
So, previously, we checked `time.Since(lastHealthy) > etcdServerOfflineTimeout`

Reply:
The endpoints returned by `MemberList` are the actual members in the raft cluster. If some member is not returned by `MemberList`, it's definitely removed from the raft cluster via config change instead of the client-side active probe. So it's safe to remove the stale endpoints that are removed from the raft cluster.

Reply:
sorry, I forgot this pr. I agree it, and it is better to run HA tests, such as network partition and pd leader io hang.